crates/misc/run-examples/src/main.rs

```diff
@@ -16,13 +16,14 @@ fn main() -> anyhow::Result<()> {
         examples.insert((path.file_stem().unwrap().to_str().unwrap().to_owned(), dir));
     }
 
-    println!("======== Building libwasmtime.a ===========");
-    run(Command::new("cargo")
-        .args(&["build"])
-        .current_dir("crates/c-api"))?;
+    println!("======== Prepare C/C++ CMake project ===========");
+    run(Command::new("cmake")
+        .arg("-Sexamples")
+        .arg("-Bexamples/build")
+        .arg("-DBUILD_SHARED_LIBS=OFF"))?;
 
     for (example, is_dir) in examples {
-        if example == "README" {
+        if example == "README" || example == "CMakeLists" || example == "build" {
```

Comment:
I pushed a solution involving `cargo read-manifest` and `cmake --build examples/build --target help`, which can be parsed to get all targets. I still need to add `--features` handling but let me know how it is

Reply:
I like it! I had tried to find an option like `cmake --target help` but didn't search for the right things, I guess.

I see you removed the `example_to_run` command-line argument. I don't know whether anyone was using that but it seems pretty easy to keep supporting it so if you don't mind putting it back that'd be cool. It's weird that it runs two examples (the C and the Rust version) most of the time, but that seems harmless.

I think the big question now is, what will CI do with this PR this time?

Reply:
Oh whoops, added it back it. It will still check if that is a valid example before it runs, however, like the old code. I'm also interested in seeing if the CI works *for once*.

Reply:
Yup, that looks right to me. And I'm interested too! :crossed_fingers: